### PR TITLE
feat: add pytrf sub-commands as individual Snakemake-Wrappers 

### PR DESCRIPTION
**Pull request by rohan-ibn-tariq**
Add individual Snakemake-Wrappers for pytrf sub-commands (findstr, findgtr, findatr, extract) to enable modular execution and parameter handling within Snakemake workflows. Each wrapper handles optional parameters, output files, and logging in a clean and minimalistic way.

### QC

* [x] I confirm that I have followed the [documentation for contributing to `snakemake-wrappers`](https://snakemake-wrappers.readthedocs.io/en/stable/contributing.html).

While the contributions guidelines are more extensive, please particularly ensure that:
* [x] `test.py` was updated to call any added or updated example rules in a `Snakefile` -> **test_wrappers.py**
* [x] `input:` and `output:` file paths in the rules can be chosen arbitrarily
* [x] wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`)
* [x] temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to -> **wrapper wasn't creating temp file, as for underlying tool I didn't changed the default behaviour**
* [x] the `meta.yaml` contains a link to the documentation of the respective tool or command under `url:`
* [x] conda environments use a minimal amount of channels and packages, in recommended ordering

## Summary by CodeRabbit

* **New Features**
  * Added PyTRF support: four CLI tools with Snakemake wrappers, metadata/manifests and demo input data.
* **Tests**
  * Added end-to-end tests for findstr, findgtr and findatr; extract test marked to skip.
* **Chores**
  * Added Conda environment manifests and pinned explicit environment snapshots for reproducible runs.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: bio/pytrf/extract/wrapper.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+# Build parameters
+params = [f"-r {repeat_file}"]
+
+try:
+    if hasattr(snakemake.params, "out_format"):
+        params.append(f"-f {snakemake.params.out_format}")
+
+    if hasattr(snakemake.params, "flank_length"):
+        params.append(f"-l {snakemake.params.flank_length}")
```
Comment (fgvieira) [inline bio/pytrf/extract/wrapper.py]: Are these parameters mandatory?

Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/wrapper.py]: Flank length and output format are not mandatory

Comment (fgvieira) [reply bio/pytrf/extract/wrapper.py]: Why the try/except? If arguments do not exist, `hasattr` just returns `False`.

Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/wrapper.py]: @fgvieira you are right, that is dead code, needs similar fix in all wrappers !
I will update that!

File: bio/pytrf/extract/environment.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+channels:
+  - conda-forge
+  - bioconda
+  - nodefaults
+
+dependencies:
+  - pytrf =1.4
+  - pyfastx =2.2
```
Comment (fgvieira) [inline bio/pytrf/extract/environment.yaml]: It should be no longer necessary, due to [PR60976](https://github.com/bioconda/bioconda-recipes/pull/60976)

```suggestion
```

Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/environment.yaml]: Thank you for resolving that out!
I will fix that!

File: bio/pytrf/extract/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+input:
+  - FASTA or FASTQ file (supports gzip compression)
+output:
+  - Output file (default -> stdout, will be redirected to the log file).
+params:
+  repeat_file: >
+    **Required.** Path to TSV or CSV file from pytrf findstr/findgtr/findatr.
+  out_format: >
+    Output format. Options: 'tsv' (default), 'csv', or 'fasta'.
+    Note: Only extract command supports FASTA output.
```
Comment (fgvieira) [inline bio/pytrf/extract/meta.yaml]: Should be inferred from output file extension.

File: bio/pytrf/extract/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+url: https://pytrf.readthedocs.io/en/latest/usage.html#commandline-interface
+authors:
+  - Muhammad Rohan Ali Asmat
+input:
+  - FASTA or FASTQ file (supports gzip compression)
+output:
+  - Output file (default -> stdout, will be redirected to the log file).
+params:
+  repeat_file: >
+    **Required.** Path to TSV or CSV file from pytrf findstr/findgtr/findatr.
```
Comment (fgvieira) [inline bio/pytrf/extract/meta.yaml]: Sounds like it shoud be an input.

File: bio/pytrf/extract/test/Snakefile [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# SAMPLE RULE: Extract tandem repeat sequences with flanking regions
+# The pytrf extract wrapper requires output from findstr, findgtr, or findatr.
+#
+# Output:
+# - If output file is specified, results are written to that file
+# - If output is omitted, pytrf writes to stdout (redirected to log file)
```
Comment (fgvieira) [inline bio/pytrf/extract/test/Snakefile]: Better to always have an output file, and use the `log` as log. This way you can always infer the output format.

File: bio/pytrf/extract/meta.yaml (hunk 3 above)
Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/meta.yaml]: Will fix that!

File: bio/pytrf/extract/meta.yaml (hunk 4 above)
Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/meta.yaml]: Was in options in official docs, but I see your point and will move there.

File: bio/pytrf/extract/test/Snakefile (hunk 5 above)
Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/test/Snakefile]: Won't this be limiting tools natural behaviour!

Comment (fgvieira) [reply bio/pytrf/extract/test/Snakefile]: For snakemake, it is good practices to always have (whenever possible) an output file, otherwise you cannot call the rule.
You can use `log`, but that should be (as the name says) for logs only.

Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/test/Snakefile]: Thanks for pointing that out, I didn't think of that!

**Comment by rohan-ibn-tariq** [comment]
> > I get your point, but do you mean to remove the custom params detection completely now and just use extra for them. Or this is also fine? Because I tried to map all available options. I mean whatever the best practice is we should adhere to that! I think both ways are used though!
> 
> If you need the parameters to some internal wrapper logic, then leave it separately. All other params can just go on extra. The script become simpler and more flexible.

@fgvieira ok I got it. First of all I think this along with your earlier bulleted instruction on keeping things minimal should be documented in contribution notes. Should I create an issue?

Secondly, I will try to fix the issues while keeping different wrappers and not uniting them to one as already agreed, with still verbose documentation related to extra param.

File: bio/pytrf/extract/test/Snakefile [hunk 6]
```diff
@@ -1,16 +1,12 @@
...
-# The pytrf extract wrapper requires output from findstr, findgtr, or findatr.
-#
-# Output:
-# - If output file is specified, results are written to that file
-# - If output is omitted, pytrf writes to stdout (redirected to log file)
+# SAMPLE RULE: Custom parameters via extra
 rule pytrf_extract:
     input:
-        "demo_data/{sample}.fasta",
+        "demo_data/{sample}.fasta",  # input[0] - reference fasta
```
Comment (fgvieira) [inline bio/pytrf/extract/test/Snakefile]: You might want to name these input files, so that the order does not matter. For example:
```
ref = "demo_data/{sample}.fasta",
repeat = "demo_data/{sample}.tsv",
```
then you can reference them in the wrapper as `snakemake.input.ref` and `snakemake.input.repeat`.

File: bio/pytrf/extract/wrapper.py [hunk 7]
```diff
@@ -6,44 +6,67 @@
...
-    raise RuntimeError(f"Parameter validation failed: {e}") from e
+ext = OUTPUT_FILE.suffix.lower()
+
+format_map = {
+    ".tsv": "tsv",
+    ".csv": "csv",
+    ".fasta": "fasta",
+    ".fa": "fasta",
+    ".fna": "fasta",
+}
```
Comment (fgvieira) [inline bio/pytrf/extract/wrapper.py]: There is also a function to infer the file format from the extension in `snakemake-wrapper-utils`:
https://github.com/snakemake/snakemake-wrapper-utils/blob/6912c9d9c9921a7f02a92940731014d87eb00080/snakemake_wrapper_utils/snakemake.py#L37-L61

File: bio/pytrf/extract/environment.yaml [hunk 8]
```diff
@@ -5,4 +5,4 @@ channels:
 
 dependencies:
   - pytrf =1.4
-  
\ No newline at end of file
+  - snakemake-wrapper-utils =0.8
```
Comment (fgvieira) [inline bio/pytrf/extract/environment.yaml]: Maybe use `snakemake-wrapper-utils =0.8.0`, so it also triggers updates on patch versions.

File: bio/pytrf/findstr/test/test.smk
Comment (fgvieira) [inline bio/pytrf/findstr/test/test.smk:1]: Why this separate file? Can't we include this rule in the `Snakefile`?

File: bio/pytrf/extract/test/Snakefile (hunk 6 above)
Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/test/Snakefile]: That's another way to handle that, maybe it is more in accordance to snakemake design guidelines.
Should I update this in issue too? But, note the flexibility still exists which should be clearly stated too.
https://github.com/snakemake/snakemake-wrappers/issues/4915

File: bio/pytrf/extract/environment.yaml (hunk 8 above)
Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/environment.yaml]: Won't your recommendation pin to the patch version, where as in the other case it will do what you intended in your comment?

File: bio/pytrf/findstr/test/test.smk
Comment (rohan-ibn-tariq) [reply bio/pytrf/findstr/test/test.smk:1]: Yes, we can! I think it is better to not hide the edge case and rather show that in docs too.

File: bio/pytrf/extract/wrapper.py (hunk 7 above)
Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/wrapper.py]: @fgvieira excuse me for not converting this to draft actually. Yes, there is but I still had in mind to test if pytrf supports all the fasta extension types as mentioned on the helper function in utils and also the fact that where pytrf docs says gff does it automatically included gff3 too?

I see you have self assigned so I won't change something for now.

File: bio/pytrf/extract/test/Snakefile (hunk 6 above)
Comment (fgvieira) [reply bio/pytrf/extract/test/Snakefile]: This is just a suggestion, but depends on your personal preference. Me, personally, I think it is a bit more clean since the name helps to describe the file, but completely up to you! 😄

File: bio/pytrf/extract/environment.yaml (hunk 8 above)
Comment (fgvieira) [reply bio/pytrf/extract/environment.yaml]: I am not sure what would be best, but it's just that all other wrappers are pinned to the patch version also.

File: bio/pytrf/extract/test/Snakefile (hunk 6 above)
Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/test/Snakefile]: fixed with named inputs!

File: bio/pytrf/extract/environment.yaml (hunk 8 above)
Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/environment.yaml]: fixed to patch version!

File: bio/pytrf/extract/wrapper.py (hunk 7 above)
Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/wrapper.py]: refactored to use the format function!

File: bio/pytrf/findstr/test/test.smk
Comment (rohan-ibn-tariq) [reply bio/pytrf/findstr/test/test.smk:1]: fixed!

File: bio/pytrf/extract/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,25 @@
+name: pytrf extract (NOT WORKING, see notes)
+description: >
+  Extract tandem repeat sequences with flanking regions from DNA sequences.
+  Requires output from pytrf findstr, findgtr, or findatr as input.
+url: https://pytrf.readthedocs.io/en/latest/usage.html#commandline-interface
+authors:
+  - Muhammad Rohan Ali Asmat
+input:
+  - FASTA or FASTQ file (supports gzip compression)
+  - Repeat File (TSV or CSV file from pytrf findstr/findgtr/findatr)
```
Comment (fgvieira) [inline bio/pytrf/extract/meta.yaml]:
```suggestion
  - ref: FASTA or FASTQ file (supports gzip compression)
  - repeat: TSV or CSV file from pytrf findstr/findgtr/findatr
```

File: bio/pytrf/environment.yaml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+channels:
+  - conda-forge
+  - bioconda
+  - nodefaults
+
+dependencies:
```
Comment (fgvieira) [inline bio/pytrf/environment.yaml:5]:
```suggestion
  - nodefaults
dependencies:
```

File: bio/pytrf/findatr/environment.yaml [hunk 11]
```diff
@@ -0,0 +1,8 @@
+channels:
+  - conda-forge
+  - bioconda
+  - nodefaults
+
+dependencies:
```
Comment (fgvieira) [inline bio/pytrf/findatr/environment.yaml:5]:
```suggestion
  - nodefaults
dependencies:
```

File: bio/pytrf/findgtr/environment.yaml [hunk 12]
```diff
@@ -0,0 +1,8 @@
+channels:
+  - conda-forge
+  - bioconda
+  - nodefaults
+
+dependencies:
```
Comment (fgvieira) [inline bio/pytrf/findgtr/environment.yaml:5]:
```suggestion
  - nodefaults
dependencies:
```

File: bio/pytrf/findstr/environment.yaml [hunk 13]
```diff
@@ -0,0 +1,8 @@
+channels:
+  - conda-forge
+  - bioconda
+  - nodefaults
+
+dependencies:
```
Comment (fgvieira) [inline bio/pytrf/findstr/environment.yaml:5]:
```suggestion
  - nodefaults
dependencies:
```

File: test_wrappers.py [hunk 14]
```diff
@@ -7189,13 +7197,14 @@ def test_orthanq(run):
             "--use-conda",
             "out/candidates",
             "out/candidates.vcf",
-#             "out/preprocess_hla.bcf",
+            #             "out/preprocess_hla.bcf",
```
Comment (fgvieira) [inline test_wrappers.py]:
```suggestion
            #"out/preprocess_hla.bcf",
```

File: test_wrappers.py [hunk 15]
```diff
@@ -7212,3 +7221,98 @@ def test_go_yq(run):
...
+            "snakemake",
+            "--cores",
+            "1",
+            "results/small_test_defaults.tsv",
+            "--use-conda",
+            "-F",
+            "--allowed-rules",
+            "pytrf_findstr_defaults",
+        ],
+    )
```
Comment (fgvieira) [inline test_wrappers.py]:
```suggestion
            "results/small_test.csv",
            "results/small_test_defaults.tsv",
        ],
        compare_results_with_expected={
            "results/small_test.csv": "expected/findstr_basic.csv",
        },
    )
```

File: test_wrappers.py [hunk 16]
```diff
@@ -7212,3 +7221,82 @@ def test_go_yq(run):
...
+        "bio/pytrf/findstr",
+        [
+            "snakemake",
+            "--cores",
+            "1",
+            "results/small_test.csv",
+            "--use-conda",
+            "-F",
+            "results/small_test.csv",
+            "results/small_test_defaults.tsv",
```
Comment (fgvieira) [inline test_wrappers.py]:
```suggestion
            "results/small_test.csv",
            "results/small_test_defaults.tsv",
            "--use-conda",
            "-F",
```

File: test_wrappers.py [hunk 17]
```diff
@@ -7212,3 +7221,81 @@ def test_go_yq(run):
...
+        "bio/pytrf/findgtr",
+        [
+            "snakemake",
+            "--cores",
+            "1",
+            "results/small_test.tsv",
+            "--use-conda",
+            "-F",
+            "--allowed-rules",
+            "pytrf_findgtr",
```
Comment (fgvieira) [inline test_wrappers.py]:
```suggestion
            "-F",
```

File: test_wrappers.py [hunk 18]
```diff
@@ -7212,3 +7221,81 @@ def test_go_yq(run):
...
+        "bio/pytrf/findatr",
+        [
+            "snakemake",
+            "--cores",
+            "1",
+            "results/small_test.tsv",
+            "--use-conda",
+            "-F",
+            "--allowed-rules",
+            "pytrf_findatr",
```
Comment (fgvieira) [inline test_wrappers.py]:
```suggestion
            "-F",
```

File: test_wrappers.py [hunk 19]
```diff
@@ -7212,3 +7221,81 @@ def test_go_yq(run):
...
+        "bio/pytrf/extract",
+        [
+            "snakemake",
+            "--cores",
+            "1",
+            "results/small_test_extract.tsv",
+            "--use-conda",
+            "-F",
+            "--allowed-rules",
+            "pytrf_extract",
```
Comment (fgvieira) [inline test_wrappers.py]:
```suggestion
            "-F",
```

File: bio/pytrf/extract/wrapper.py [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+
+# Build command
+CMD = f"pytrf extract {input_file} -r {repeat_file} -f {out_format}"
+if extra:
+    CMD += f" {extra}"
+CMD += f" -o {output_file}"
+
+# Execute
+try:
+    shell(f"{CMD} {log}")
```
Comment (fgvieira) [inline bio/pytrf/extract/wrapper.py]:
```suggestion
# Execute
try:
    shell("pytrf extract"
          " {input_file}"
          " -r {repeat_file}"
          " {extra}"
          " -f {out_format}"
          " -o {output_file}"
          " {log}"
    )
```

File: bio/pytrf/findatr/wrapper.py [hunk 21]
```diff
@@ -0,0 +1,55 @@
...
+    )
+
+CMD = f"pytrf findatr {input_file} -f {out_format}"
+if extra:
+    CMD += f" {extra}"
+CMD += f" -o {output_file}"
+
+# Execute
+try:
+    shell(f"{CMD} {log}")
```
Comment (fgvieira) [inline bio/pytrf/findatr/wrapper.py]:
```suggestion
# Execute
try:
    shell("pytrf findatr"
          " {input_file}"
          " {extra}"
          " -f {out_format}"
          " -o {output_file}"
          " {log}"
    )
```

File: bio/pytrf/findgtr/wrapper.py [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+
+# Build command
+CMD = f"pytrf findgtr {input_file} -f {out_format}"
+if extra:
+    CMD += f" {extra}"
+CMD += f" -o {output_file}"
+
+# Execute
+try:
+    shell(f"{CMD} {log}")
```
Comment (fgvieira) [inline bio/pytrf/findgtr/wrapper.py]:
```suggestion
# Execute
try:
    shell("pytrf findgtr"
          " {input_file}"
          " {extra}"
          " -f {out_format}"
          " -o {output_file}"
          " {log}"
    )
```

File: bio/pytrf/extract/wrapper.py [hunk 23]
```diff
@@ -0,0 +1,67 @@
...
+
+# Execute
+try:
+    shell("pytrf extract"
+          " {input_file}"
+          " -r {repeat_file}"
+          " {extra}"
+          " -f {out_format}"
+          " -o {output_file}"
+          " {log}"
```
Comment (fgvieira) [inline bio/pytrf/extract/wrapper.py]:
```suggestion
        " {input_file}"
        " -r {repeat_file}"
        " {extra}"
        " -f {out_format}"
        " -o {output_file}"
        " {log}"
```

File: bio/pytrf/findatr/wrapper.py [hunk 24]
```diff
@@ -0,0 +1,56 @@
...
+    )
+
+# Execute
+try:
+    shell("pytrf findatr"
+          " {input_file}"
+          " {extra}"
+          " -f {out_format}"
+          " -o {output_file}"
+          " {log}"
```
Comment (fgvieira) [inline bio/pytrf/findatr/wrapper.py]:
```suggestion
        " {input_file}"
        " {extra}"
        " -f {out_format}"
        " -o {output_file}"
        " {log}"
```

File: bio/pytrf/findgtr/wrapper.py [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+    )
+
+# Execute
+try:
+    shell("pytrf findgtr"
+          " {input_file}"
+          " {extra}"
+          " -f {out_format}"
+          " -o {output_file}"
+          " {log}"
```
Comment (fgvieira) [inline bio/pytrf/findgtr/wrapper.py]:
```suggestion
        " {input_file}"
        " {extra}"
        " -f {out_format}"
        " -o {output_file}"
        " {log}"
```

File: bio/pytrf/extract/wrapper.py [hunk 26]
```diff
@@ -0,0 +1,67 @@
...
+if is_arg("-r", extra) or is_arg("--repeat-file", extra):
+    raise ValueError(
+        "Repeat file should be specified as an input. "
+        "Do not specify -r/--repeat-file in params.extra"
+    )
+
+# Execute
+try:
+    shell("pytrf extract"
+        " {input_file}"
```
Comment (fgvieira) [inline bio/pytrf/extract/wrapper.py]:
```suggestion
    shell(
        "pytrf extract"
        " {input_file}"
```

File: bio/pytrf/findatr/wrapper.py [hunk 27]
```diff
@@ -0,0 +1,56 @@
...
+
+if is_arg("-f", extra) or is_arg("--out-format", extra):
+    raise ValueError(
+        "Output format is automatically inferred from output file extension. "
+        "Do not specify -f/--out-format in params.extra. "
+    )
+
+# Execute
+try:
+    shell("pytrf findatr"
```
Comment (fgvieira) [inline bio/pytrf/findatr/wrapper.py]:
```suggestion
    shell(
        "pytrf findatr"
```

File: bio/pytrf/findgtr/wrapper.py [hunk 28]
```diff
@@ -0,0 +1,56 @@
...
+
+if is_arg("-f", extra) or is_arg("--out-format", extra):
+    raise ValueError(
+        "Output format is automatically inferred from output file extension. "
+        "Do not specify -f/--out-format in params.extra. "
+    )
+
+# Execute
+try:
+    shell("pytrf findgtr"
```
Comment (fgvieira) [inline bio/pytrf/findgtr/wrapper.py]:
```suggestion
    shell(
        "pytrf findgtr"
```

File: bio/pytrf/extract/wrapper.py [hunk 29]
```diff
@@ -0,0 +1,68 @@
...
+
+out_format = get_format(output_file)
+
+supported_formats = {"tsv", "csv", "fasta"}
+
+if out_format not in supported_formats:
+    raise ValueError(
+        f"Unsupported format '{out_format}' for pytrf extract. "
+        f"Supported formats: {', '.join(supported_formats)}"
+    )
```
[the remark this answers is not in the document]
Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/wrapper.py]: getformat will return fasta for multiple supported fasta format types so this is not valid right? #coderabbit

[the remark this answers is not in the document]
Comment (rohan-ibn-tariq) [reply bio/pytrf/extract/wrapper.py]: can you resolve this?

File: bio/pytrf/findgtr/wrapper.py [hunk 30]
```diff
@@ -0,0 +1,56 @@
...
+    )
+
+# Additional parameters
+extra = snakemake.params.get("extra", "")
+
+if is_arg("-f", extra) or is_arg("--out-format", extra):
+    raise ValueError(
+        "Output format is automatically inferred from output file extension. "
+        "Do not specify -f/--out-format in params.extra. "
+    )
```
[the remark this answers is not in the document]
Comment (rohan-ibn-tariq) [reply bio/pytrf/findgtr/wrapper.py]: fixed in latest 4 commits!

File: bio/pytrf/test/extra_rules.smk
Comment (fgvieira) [inline bio/pytrf/test/extra_rules.smk:1]: Why not include all these rules in the `Snakefile`?

File: bio/pytrf/wrapper.py [hunk 31]
```diff
@@ -0,0 +1,108 @@
...
+log = snakemake.log_fmt_shell(stdout=True, stderr=True)
+
+# Get subcommand type and validate
+subcommand = snakemake.params.get("subcommand")
+if not subcommand:
+    raise ValueError(
+        "params.subcommand is required. "
+        f"Valid options: {', '.join(sorted(VALID_SUBCOMMANDS))}"
+    )
+if subcommand not in VALID_SUBCOMMANDS:
```
Comment (fgvieira) [inline bio/pytrf/wrapper.py]:
```suggestion
if subcommand is None or subcommand not in VALID_SUBCOMMANDS:
```

File: bio/pytrf/test/extra_rules.smk
Comment (rohan-ibn-tariq) [reply bio/pytrf/test/extra_rules.smk:1]: Actually, extract command was not working for now.
As for other two I find them too similar in usage to findstr that I thought to just keep the rules for test not for documentation, because documentation is quite elaborate already.

File: bio/pytrf/meta.yaml [hunk 32]
```diff
@@ -0,0 +1,42 @@
...
+  - repeat: For extract only - TSV/CSV file from findstr/findgtr/findatr
+output:
+  - Output file (required). Format auto-detected from file extension.
+params:
+  subcommand: >
+    Required. PyTRF subcommand to run.
+    Options: findstr, findgtr, findatr, extract.
+  extra: >
+    Additional command-line arguments passed to pytrf.
+    See url for subcommand-specific options.
```
Comment (fgvieira) [inline bio/pytrf/meta.yaml]:
```suggestion
  - subcommand: PyTRF subcommand to run [`findstr`, `findgtr`, `findatr`, `extract`].
  - extra: Additional command-line arguments passed to pytrf.
```

File: bio/pytrf/test/extra_rules.smk
Comment (fgvieira) [reply bio/pytrf/test/extra_rules.smk:1]: But these rules can also be used, no? If so, it would make sense to have them in the docs also.

Comment (rohan-ibn-tariq) [reply bio/pytrf/test/extra_rules.smk:1]: yes, they can be used apart from extract for now.

Comment (fgvieira) [reply bio/pytrf/test/extra_rules.smk:1]: So just put them all in the `Snakefile`, so it show up in the docs.

Comment (rohan-ibn-tariq) [reply bio/pytrf/test/extra_rules.smk:1]: ok

**Review by fgvieira — APPROVED** [review]

**Comment by fgvieira** [comment]
Nice work @rohan-ibn-tariq ! :+1:

**Comment by rohan-ibn-tariq** [comment]
> Nice work @rohan-ibn-tariq ! 👍

Thanks for your time, assistance and contributions too! @fgvieira
